I still don't understand how I create a pagination

@RequiredArgsConstructor
public class FriendsView extends View {


    private final LobbyPlugin plugin;
    private State<Pagination> paginationState;


    @Override
    public void onInit(ViewConfigBuilder config) {
        config.title("Lista de amigos")
                .size(6)
                .layout(
                        "         ",
                        " OOOOOOO ",
                        " OOOOOOO ",
                        "         ",
                        "         ",
                        "         ")
                .cancelOnClick();
    }

    @Override
    public void onFirstRender(RenderContext render) {
        this.paginationState = paginationState(
                listFriends(render.getPlayer()),
                (item, friends) -> item.withItem(
                        new ItemBuilder(Material.SKULL_ITEM)
                                .name("")
                                .build()
                )
        );
    }


    private List<Friends> listFriends(Player player) {
        return plugin.getUserCache().getByName(player.getName()).getFriends();
    }
    
    
    ERROR: View is already initialized, please move this method call to class construtor or #onInit.

ERROR: ```[19:23:53] [Server thread/ERROR]: Could not pass event InventoryClickEvent to spigot-spawnershop v1.0
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched_1.8.8.jar:git-PaperSpigot-445]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78) ~[patched_1.8.8.jar:git-PaperSpigot-445]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.8.8.jar:git-PaperSpigot-445]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:517) [patched_1.8.8.jar:git-PaperSpigot-445]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:502) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1657) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.PacketPlayInWindowClick.a(SourceFile:31) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.PacketPlayInWindowClick.a(SourceFile:9) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched_1.8.8.jar:git-PaperSpigot-445]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_362]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_362]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched_1.8.8.jar:git-PaperSpigot-445]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched_1.8.8.jar:git-PaperSpigot-445]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_362]
Caused by: java.lang.NullPointerException
	at net.hyren.spawnershop.view.StoreView.item(StoreView.java:93) ~[?:?]
	at net.hyren.spawnershop.view.StoreView.onFirstRender(StoreView.java:51) ~[?:?]
	at net.hyren.spawnershop.view.StoreView.onFirstRender(StoreView.java:22) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PlatformRenderInterceptor.intercept(PlatformRenderInterceptor.java:16) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PlatformRenderInterceptor.intercept(PlatformRenderInterceptor.java:8) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.loop(PipelineContext.java:43) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.proceed(PipelineContext.java:53) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.execute(PipelineContext.java:59) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.Pipeline.execute(Pipeline.java:112) ~[?:?]
	at net.hyren.core.spigot.inventory.PlatformView.renderContext(PlatformView.java:295) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PlatformOpenInterceptor.finishOpen(PlatformOpenInterceptor.java:54) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PlatformOpenInterceptor.intercept(PlatformOpenInterceptor.java:28) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PlatformOpenInterceptor.intercept(PlatformOpenInterceptor.java:12) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.loop(PipelineContext.java:43) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.proceed(PipelineContext.java:53) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.execute(PipelineContext.java:59) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.Pipeline.execute(Pipeline.java:119) ~[?:?]
	at net.hyren.core.spigot.inventory.PlatformView.open(PlatformView.java:132) ~[?:?]
	at net.hyren.core.spigot.inventory.PlatformView.navigateTo(PlatformView.java:160) ~[?:?]
	at net.hyren.core.spigot.inventory.context.PlatformConfinedContext.openForPlayer(PlatformConfinedContext.java:25) ~[?:?]
	at net.hyren.core.spigot.inventory.context.PlatformRenderContext.openForPlayer(PlatformRenderContext.java:316) ~[?:?]
	at net.hyren.core.spigot.inventory.context.SlotClickContext.openForPlayer(SlotClickContext.java:150) ~[?:?]
	at net.hyren.spawnershop.view.SpawnersView.lambda$null$0(SpawnersView.java:67) ~[?:?]
	at net.hyren.core.spigot.inventory.component.ItemComponent.clicked(ItemComponent.java:204) ~[?:?]
	at net.hyren.core.spigot.inventory.component.PaginationImpl.clicked(PaginationImpl.java:634) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.ItemClickInterceptor.intercept(ItemClickInterceptor.java:35) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.ItemClickInterceptor.intercept(ItemClickInterceptor.java:14) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.loop(PipelineContext.java:43) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.proceed(PipelineContext.java:53) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.PipelineContext.execute(PipelineContext.java:59) ~[?:?]
	at net.hyren.core.spigot.inventory.pipeline.Pipeline.execute(Pipeline.java:119) ~[?:?]
	at net.hyren.core.spigot.inventory.IFInventoryListener.onInventoryClick(IFInventoryListener.java:57) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_362]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_362]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_362]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_362]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:300) ~[patched_1.8.8.jar:git-PaperSpigot-445]
	... 16 more```

--- a/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfig.java
+++ b/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfig.java
@@ -29,6 +29,7 @@ public class ViewConfig {
     private final String[] layout;
     private final Set<Modifier> modifiers;
     private final long updateIntervalInTicks, interactionDelayInMillis;
+    private final boolean transitiveInitialData;
 
     public ViewConfig(
             Object title,
@@ -38,7 +39,8 @@ public class ViewConfig {
             String[] layout,
             Set<Modifier> modifiers,
             long updateIntervalInTicks,
-            long interactionDelayInMillis) {
+            long interactionDelayInMillis,
+            boolean transitiveInitialData) {
         this.title = title;
         this.size = size;
         this.type = type;
@@ -47,6 +49,7 @@ public class ViewConfig {
         this.modifiers = modifiers;
         this.updateIntervalInTicks = updateIntervalInTicks;
         this.interactionDelayInMillis = interactionDelayInMillis;
+        this.transitiveInitialData = transitiveInitialData;
     }
 
     public Object getTitle() {
@@ -79,6 +82,10 @@ public class ViewConfig {
 
     public long getInteractionDelayInMillis() {
         return interactionDelayInMillis;
+    }
+
+    public boolean isTransitiveInitialData() {
+        return transitiveInitialData;
     }
 
     @VisibleForTesting
@@ -138,7 +145,8 @@ public class ViewConfig {
                 merge(other, ViewConfig::getLayout),
                 merge(other, ViewConfig::getModifiers, value -> value != null && !value.isEmpty()),
                 merge(other, ViewConfig::getUpdateIntervalInTicks, value -> value != 0),
-                merge(other, ViewConfig::getInteractionDelayInMillis, value -> value != 0));
+                merge(other, ViewConfig::getInteractionDelayInMillis, value -> value != 0),
+                merge(other, ViewConfig::isTransitiveInitialData));
     }
 
     private <T> T merge(ViewConfig other, Function<ViewConfig, T> retriever) {
@@ -228,7 +236,8 @@ public class ViewConfig {
                 && Objects.equals(getType(), that.getType())
                 && Objects.equals(getOptions(), that.getOptions())
                 && Arrays.equals(getLayout(), that.getLayout())
-                && Objects.equals(getModifiers(), that.getModifiers());
+                && Objects.equals(getModifiers(), that.getModifiers())
+                && isTransitiveInitialData() == that.isTransitiveInitialData();
     }
 
     @Override
@@ -240,7 +249,8 @@ public class ViewConfig {
                 getOptions(),
                 getModifiers(),
                 getUpdateIntervalInTicks(),
-                getInteractionDelayInMillis());
+                getInteractionDelayInMillis(),
+                isTransitiveInitialData());
         result = 31 * result + Arrays.hashCode(getLayout());
         return result;
     }
@@ -255,6 +265,7 @@ public class ViewConfig {
                 + Arrays.toString(layout) + ", modifiers="
                 + modifiers + ", updateIntervalInTicks="
                 + updateIntervalInTicks + ", interactionDelayInMillis="
-                + interactionDelayInMillis + '}';
+                + interactionDelayInMillis + ", transitiveInitialData="
+                + transitiveInitialData + "}";
     }
 }

--- a/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfigBuilder.java
+++ b/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfigBuilder.java
@@ -227,7 +227,7 @@ public final class ViewConfigBuilder {
                 getModifiers(),
                 getUpdateIntervalInTicks(),
                 getInteractionDelayInMillis(),
-			transitiveInitialData);
+                transitiveInitialData);
     }
 
     public static boolean isTitleAsComponentSupported() {
@@ -266,7 +266,7 @@ public final class ViewConfigBuilder {
         return interactionDelayInMillis;
     }
 
-	public boolean isTransitiveInitialData() {
-		return transitiveInitialData;
-	}
+    public boolean isTransitiveInitialData() {
+        return transitiveInitialData;
+    }
 }

--- a/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfigBuilder.java
+++ b/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfigBuilder.java
@@ -31,6 +31,7 @@ public final class ViewConfigBuilder {
     private String[] layout = null;
     private final Set<ViewConfig.Modifier> modifiers = new HashSet<>();
     private long updateIntervalInTicks, interactionDelayInMillis;
+    private boolean transitiveData;
 
     /**
      * Inherits all configuration from another {@link ViewConfigBuilder} value.
@@ -193,6 +194,21 @@ public final class ViewConfigBuilder {
     @ApiStatus.Experimental
     public ViewConfigBuilder interactionDelay(Duration interactionDelay) {
         this.interactionDelayInMillis = interactionDelay == null ? 0 : interactionDelay.toMillis();
+        return this;
+    }
+
+    /**
+     * When navigating between views data from "A" to "B" is not carried so trying to access
+     * some data from "A" in "B" will throw a {@link NullPointerException}.
+     * <p>
+     * This behavior can be changed by enabling this option, once enabled, on every navigation
+     * between views data will be carried from the view where came from to the view where are going.
+     *
+     * @return This configuration builder.
+     * @see <a href="https://github.com/DevNatan/inventory-framework/wiki/navigating-between-views">Navigating Between Views on Wiki</a>
+     */
+    public ViewConfigBuilder transitiveInitialData(boolean transitiveInitialData) {
+        this.transitiveData = transitiveInitialData;
         return this;
     }
 

--- a/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfigBuilder.java
+++ b/inventory-framework-api/src/main/java/me/devnatan/inventoryframework/ViewConfigBuilder.java
@@ -31,7 +31,7 @@ public final class ViewConfigBuilder {
     private String[] layout = null;
     private final Set<ViewConfig.Modifier> modifiers = new HashSet<>();
     private long updateIntervalInTicks, interactionDelayInMillis;
-    private boolean transitiveData;
+    private boolean transitiveInitialData;
 
     /**
      * Inherits all configuration from another {@link ViewConfigBuilder} value.
@@ -208,7 +208,7 @@ public final class ViewConfigBuilder {
      * @see <a href="https://github.com/DevNatan/inventory-framework/wiki/navigating-between-views">Navigating Between Views on Wiki</a>
      */
     public ViewConfigBuilder transitiveInitialData(boolean transitiveInitialData) {
-        this.transitiveData = transitiveInitialData;
+        this.transitiveInitialData = transitiveInitialData;
         return this;
     }
 
@@ -226,7 +226,8 @@ public final class ViewConfigBuilder {
                 getLayout(),
                 getModifiers(),
                 getUpdateIntervalInTicks(),
-                getInteractionDelayInMillis());
+                getInteractionDelayInMillis(),
+			transitiveInitialData);
     }
 
     public static boolean isTitleAsComponentSupported() {
@@ -264,4 +265,8 @@ public final class ViewConfigBuilder {
     long getInteractionDelayInMillis() {
         return interactionDelayInMillis;
     }
+
+	public boolean isTransitiveInitialData() {
+		return transitiveInitialData;
+	}
 }

--- a/inventory-framework-platform/src/main/java/me/devnatan/inventoryframework/context/PlatformConfinedContext.java
+++ b/inventory-framework-platform/src/main/java/me/devnatan/inventoryframework/context/PlatformConfinedContext.java
@@ -1,5 +1,10 @@
 package me.devnatan.inventoryframework.context;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import me.devnatan.inventoryframework.RootView;
 import me.devnatan.inventoryframework.Viewer;
 import org.jetbrains.annotations.NotNull;
@@ -16,13 +21,49 @@ abstract class PlatformConfinedContext extends PlatformContext implements IFConf
 
     @Override
     public void openForPlayer(@NotNull Class<? extends RootView> other) {
-        openForPlayer(other, null);
+        openForPlayer(other, getConfig().isTransitiveInitialData() ? getInitialData() : null);
     }
 
     @SuppressWarnings("unchecked")
     @Override
     public void openForPlayer(@NotNull Class<? extends RootView> other, Object initialData) {
-        getRoot().navigateTo(other, (IFRenderContext) this, getViewer(), initialData);
+        getRoot()
+                .navigateTo(
+                        other,
+                        (IFRenderContext) this,
+                        getViewer(),
+                        getConfig().isTransitiveInitialData() ? mergeInitialData(initialData) : initialData);
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private Object mergeInitialData(Object otherData) {
+        final Object initialData = getInitialData();
+        if (initialData == null || otherData == null) return otherData == null ? initialData : otherData;
+
+        if (!(initialData.getClass().equals(otherData.getClass())))
+            throw new IllegalArgumentException(String.format(
+                    "Failed to merge initial data, supplied data and current initial data must have the same type."
+                            + " Initial data type is \"%s\" and supplied data type is \"%s\". Try changing type of both to a similar type like Map in order to merge be executed.",
+                    initialData.getClass().getName(), otherData.getClass().getName()));
+
+        final Object resultData;
+        if (initialData instanceof List && otherData instanceof List) {
+            final List<?> newData = new ArrayList<>();
+            newData.addAll((List) initialData);
+            newData.addAll((List) otherData);
+            resultData = Collections.unmodifiableList(newData);
+        } else if (initialData instanceof Map && otherData instanceof Map) {
+            final Map<?, ?> newData = new HashMap<>();
+            newData.putAll((Map) initialData);
+            newData.putAll((Map) otherData);
+            resultData = Collections.unmodifiableMap(newData);
+        } else {
+            throw new IllegalArgumentException(String.format(
+                    "Unable to merge initial data (initial data type = %s, supplied data type = %s)",
+                    initialData.getClass().getName(), otherData.getClass().getName()));
+        }
+
+        return resultData;
     }
 
     @Override

--- a/inventory-framework-platform/src/main/java/me/devnatan/inventoryframework/context/PlatformConfinedContext.java
+++ b/inventory-framework-platform/src/main/java/me/devnatan/inventoryframework/context/PlatformConfinedContext.java
@@ -21,18 +21,19 @@ abstract class PlatformConfinedContext extends PlatformContext implements IFConf
 
     @Override
     public void openForPlayer(@NotNull Class<? extends RootView> other) {
-        openForPlayer(other, getConfig().isTransitiveInitialData() ? getInitialData() : null);
+        openForPlayer(other, getConfig().isTransitiveInitialData() ? getInitialData() : null, false);
+    }
+
+    @Override
+    public void openForPlayer(@NotNull Class<? extends RootView> other, Object initialData) {
+        openForPlayer(other, initialData, true);
     }
 
     @SuppressWarnings("unchecked")
-    @Override
-    public void openForPlayer(@NotNull Class<? extends RootView> other, Object initialData) {
-        getRoot()
-                .navigateTo(
-                        other,
-                        (IFRenderContext) this,
-                        getViewer(),
-                        getConfig().isTransitiveInitialData() ? mergeInitialData(initialData) : initialData);
+    private void openForPlayer(@NotNull Class<? extends RootView> other, Object initialData, boolean mergeInitialData) {
+        final Object data =
+                getConfig().isTransitiveInitialData() && mergeInitialData ? mergeInitialData(initialData) : initialData;
+        getRoot().navigateTo(other, (IFRenderContext) this, getViewer(), data);
     }
 
     @SuppressWarnings({"unchecked", "rawtypes"})

